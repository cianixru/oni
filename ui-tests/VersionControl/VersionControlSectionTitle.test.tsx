import * as React from "react"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"

import VersionControlTitle, {
    Title,
} from "./../../browser/src/UI/components/VersionControl/SectionTitle"

describe("<VersionControlTitle />", () => {
    it("correctly renders without crashing", () => {
        const wrapper = shallow(
            <VersionControlTitle
                active
                isSelected
                title="test"
                onClick={() => ({})}
                testId="test"
            />,
        )

        expect(wrapper.length).toBe(1)
    })

    it("should render the correct title", () => {
        const wrapper = shallow(
            <VersionControlTitle
                active
                isSelected
                title="test"
                onClick={() => ({})}
                testId="test"
            />,
        )
            .dive()
            .find(Title)
            .dive()
        expect(wrapper.text()).toBe("TEST")
    })

    it("Should correctly show the count if present", () => {
        const wrapper = shallow(
            <VersionControlTitle
                active
                isSelected
                title="test"
                onClick={() => ({})}
                count={5}
                testId="test"
            />,
        )

        expect(
            wrapper
                .dive()
                .find("strong")
                .text(),
        ).toBe("5")
    })

    it("should match last known snapshot", () => {
        const wrapper = shallow(
            <VersionControlTitle
                active
                isSelected
                title="test"
                onClick={() => ({})}
                testId="test"
            />,
        )
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })
})
