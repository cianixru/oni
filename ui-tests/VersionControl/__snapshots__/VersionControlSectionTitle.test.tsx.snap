// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VersionControlTitle /> should match last known snapshot 1`] = `
<styled.div
  data-test="test"
  isSelected={true}
  onClick={[Function]}
>
  <Caret
    active={true}
  />
  <styled.h4>
    TEST
  </styled.h4>
  <strong />
</styled.div>
`;
